src/validation.cpp

```diff
@@ -4706,6 +4706,11 @@ bool Chainstate::LoadChainTip()
               m_chain.Height(),
               FormatISO8601DateTime(tip->GetBlockTime()),
               GuessVerificationProgress(m_chainman.GetParams().TxData(), tip));
+
+    // Ensure KernelNotifications m_tip_block is set even if no new block arrives.
+    // Ignoring return value for now.
+    (void)m_chainman.GetNotifications().blockTip(GetSynchronizationState(/*init=*/true, m_chainman.m_blockman.m_blockfiles_indexed), *pindex);
```

Comment:
`LoadChainTip()` is called in a loop for all chainstates in `CompleteChainstateInitialization()` so that it would be called twice, for different blocks, in the case of an ongoing `AssumeUtxo` sync.
This seems problematic to me - how would subscribers to `blockTip` be supposed to deal with that? I think that the background chainstate shouldn't send a notification, as it it is done in `ActivateBestChain()`.

Reply:
> `LoadChainTip()` is called in a loop for all chainstates in `CompleteChainstateInitialization()` so that `blockTip` would be called twice, for different blocks, in the case of an ongoing `AssumeUtxo` sync. This seems problematic to me - how would subscribers to `blockTip` be supposed to deal with that? I think that the background chainstate shouldn't send a notification, as it it is done in `ActivateBestChain()`.